website/docs/advanced-concepts/access-control/roles.md

```diff
@@ -0,0 +1,247 @@
+You can't create queries for a datasource by giving create access for the datasources in the workspace. To create new queries, check the create permissions under [Application resources permissions](#app-resources-permissions) 
+
+#### Edit 
+
+**Edit** allows the user to change/modify the existing datasources. You can edit the datasource name, configurations and further test and save the datasource. Edit permission can be given at different hierarchical levels -
+
+##### Workspace
+
+If you give edit permission to the workspace, it gives access to modify all the datasources in the respective workspace. This also automatically enables permission to edit, view, all the datasources in the workspace.
+
+
+##### Datasource
+
+If you want the user to have edit access only to a specific datasource in a workspace, open the respective workspace dropdown and check the edit permission adjacent to the respective datasource. **Edit** for just a datasource allows you to modify the name, configurations of that datasource.
+
+#### View
+
+**View** only allows a user to have view access for the existing datasources. The user can interact with the functionality of the datasource, test the datasource, but can't make any changes with View permission. View permission can be given at different hierarchical levels -
+
+##### Workspace
+
+If you give view permission to the workspace, it gives access to view and execute all the datasources in the respective workspace.
+
+##### Datasource 
+
+If you want the user to have view access only to a specific datasource in a workspace, open the respective workspace's dropdown and check the view permission adjacent to the respective datasource. **View** for just a datasource allows you to view, test and execute that datasource.
+
+#### Delete
+
+**Delete** gives the user an access to delete the existing datasources in a workspace. With the delete permission, one can't edit or create a datasource, but, the user has the access to delete it.
+Delete permission can be given at various hierarchical levels - 
+
+##### Workspace
+
+If you give delete permission to the workspace, It gives access to remove any datasource from the respective workspace. 
+
+##### Datasource
+
+**Delete** for a particular datasource allows you to remove that datasource only.
+
+### Groups and roles permissions
+
+For Groups and roles for an Appsmith instance, here are the permissions that can be assigned to user or a group -
+
+#### Create
+
+With this permission, users can create new groups and roles. 
+
+##### Groups
+
+When the create permission is enabled for groups, a user can create a new group. This also automatically enables permission to view, edit, delete, invite, or remove a user from an existing group.
+
+##### Roles
+
+With create permission enables for roles, a user can create a new role. This also automatically enables permission to view, edit, delete, the existing roles.
+
```

Comment:
Also enables the associate role permission

Reply:
Removing the UX impact as discussed on the call.